### PR TITLE
feature/1.19.0

### PR DESCRIPTION
**Pull request by abeheron1**
PR to merge java artefacts 1.19 CDS changes to master.

### CODE REVIEW

**Review by vadkor — CHANGES_REQUESTED** [review]

File: client-cli/keystore/keystore.jwks [hunk 1]
```diff
@@ -14,4 +14,4 @@
       "n": "jpKqpCQkE8SzjdujeKMJo2W6x20UM3MRmmTzBAIlQKtRpAZXlALGKnQ89kpSNBc4ejuWNXRLgYwHhma6La70zhicrvsG6Cdeld9172NqhFCsedbapp7XMfJ6gjM7jScyjxW6_1emqJLI8eMLC0SCLAq9crAPv5cWcdWyapaN9n3XvPbkG9iPis6xVX3mKrC5DZ6hcOwmgsgTw92qH2i3XrixL3iQKoXKbCCKOfEWTB4aH9SBYJcpBmMALGRDogbHNFTpK0I7uGE1mxaPnEttsRf7u9yZsrfKwTZnE323w05F6r-fzcr0PZpAWGNH02vRBAkGaOxEv2Btg3UWR2JEfQ"
     }
   ]
-}
```
[the remark this answers is not in the document]
Comment (abeheron1) [reply client-cli/keystore/keystore.jwks:17]: Didn't intend, likely just whitespace while looking at it. Will undo.

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountV2.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    private String accountId;
+    private String accountNumber;
+    private String displayName;
+
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
+    @JsonFormat(shape = JsonFormat.Shape.STRING)
+    private LocalDate creationDate;
+
+    @Valid
+    @Transient
```
[the remark this answers is not in the document]
Comment (abeheron1) [reply data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountV2.java:35]: Yes, I didn't like it either. But short of actually having any plans to map in the json files. I'll see if I can somehow put some sample data and thus remove the transient.

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountV1.java [hunk 3]
```diff
@@ -24,9 +29,11 @@ public class EnergyAccount implements EnergyAccountBase {
     private LocalDate creationDate;
 
     @Valid
+    @Transient
```
[the remark this answers is not in the document]
Comment (abeheron1) [reply data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountV1.java]: Fixed.

File: client-cli/src/main/java/au/org/consumerdatastandards/client/cli/energy/EnergyAccounts.java [hunk 4]
```diff
@@ -36,14 +26,16 @@ public class EnergyAccounts extends ApiCliBase {
...
             @ShellOption(defaultValue = ShellOption.NULL) Integer page,
             @ShellOption(defaultValue = ShellOption.NULL) Integer pageSize) throws Exception {
 
         LOGGER.info("List Energy accounts CLI initiated with page: {}, page-size: {}", page, pageSize);
 
         api.setApiClient(clientFactory.create(true, check));
-        ApiResult<EnergyAccountListResponse> result = api.listEnergyAccounts(page, pageSize);
-        ApiResponse<EnergyAccountListResponse> response = result.getResponse();
+        ApiResult<EnergyAccountListResponse<EnergyAccountV1>> result = api.listEnergyAccounts(openStatus, version, page, pageSize);
+        ApiResponse<EnergyAccountListResponse<EnergyAccountV1>> response = result.getResponse();
```
[the remark this answers is not in the document]
Comment (abeheron1) [reply client-cli/src/main/java/au/org/consumerdatastandards/client/cli/energy/EnergyAccounts.java:51]: Should really be EnergyAccountBase, corrected.

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/util/CdsDataLoader.java [hunk 5]
```diff
@@ -99,6 +108,9 @@ public void loadAll() throws IOException {
         load("payloads/banking/transactions", transactionDetailRepository, BankingTransactionDetail.class);
 
         // Energy
+        //load("payloads/energy/accounts", energyAccountV1Repository, EnergyAccountV2.class);
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/util/CdsDataLoader.java:111]: Forgotten to remove this line?

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountDetailAuthorisedContacts.java [hunk 6]
```diff
@@ -11,12 +18,22 @@
 /**
  * EnergyAccountDetailAllOfAuthorisedContacts
  */
+@Entity
+@Table(name = "AuthorisedContact")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountDetailAuthorisedContacts.java:22]: Maybe better EnergyAuthorisedContact?

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContract.java [hunk 7]
```diff
@@ -11,8 +21,16 @@
 /**
  * EnergyPlanContract
  */
+@Entity
+@Table(name = "EnergyPlanContract")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContract.java:25]: No need for this line

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountPlans.java [hunk 8]
```diff
@@ -11,12 +20,23 @@
 /**
  * EnergyAccountAllOfPlans
  */
+@Entity
+@Table(name = "EnergyAccountPlan")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountPlans.java:24]: No need for this line

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountPlanOverview.java [hunk 9]
```diff
@@ -1,14 +1,29 @@
...
+import javax.persistence.Id;
+import javax.persistence.Table;
 import javax.validation.constraints.NotNull;
 import java.util.Objects;
 
 /**
  * EnergyAccountAllOfPlanOverview
  */
+@Entity
+@Table(name = "PlanOverview")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountPlanOverview.java:18]: Better: EnergyPlanOverview - other industries will have Plans

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountDetailPlans.java [hunk 10]
```diff
@@ -11,17 +21,30 @@
 /**
  * EnergyAccountDetailAllOfPlans
  */
+@Entity
+@Table(name="DetailPlan")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountDetailPlans.java:25]: EnergyDetailPlan?

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractDiscounts.java [hunk 11]
```diff
@@ -17,6 +18,7 @@
  * EnergyPlanContractDiscounts
  */
 @Entity
+@Table(name="ContractDiscount")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractDiscounts.java:21]: EnergyContractDiscount?

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractEligibility.java [hunk 12]
```diff
@@ -7,13 +7,15 @@
...
 import javax.persistence.Id;
+import javax.persistence.Table;
 import javax.validation.constraints.NotNull;
 import java.util.Objects;
 
 /**
  * EnergyPlanContractEligibility
  */
 @Entity
+@Table(name="ContractEligibility")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractEligibility.java:18]: EnergyContractEligibility?

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractFees.java [hunk 13]
```diff
@@ -9,13 +9,15 @@
...
 import javax.persistence.Id;
+import javax.persistence.Table;
 import javax.validation.constraints.NotNull;
 import java.util.Objects;
 
 /**
  * EnergyPlanContractFees
  */
 @Entity
+@Table(name="ContractFee")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractFees.java:20]: EnergyContractFee?

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractIncentives.java [hunk 14]
```diff
@@ -7,13 +7,15 @@
...
 import javax.persistence.Id;
+import javax.persistence.Table;
 import javax.validation.constraints.NotNull;
 import java.util.Objects;
 
 /**
  * EnergyPlanContractIncentives
  */
 @Entity
+@Table(name="ContractIncentive")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractIncentives.java:18]: EnergyContractIncentive?

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractTariffPeriod.java [hunk 15]
```diff
@@ -22,6 +23,7 @@
  * EnergyPlanContractTariffPeriod
  */
 @Entity
+@Table(name="TariffPeriod")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyPlanContractTariffPeriod.java:26]: EnergyTariffPeriod?

File: client/src/main/java/au/org/consumerdatastandards/client/api/ProtectedAPI.java [hunk 16]
```diff
@@ -4,14 +4,16 @@
...
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.Map;
 
 public class ProtectedAPI extends UnprotectedAPI {
 
     protected void addCdsProtectedApiHeaders(Map<String, String> headers) {
         headers.put("x-fapi-customer-ip-address", getIpAddress());
-        headers.put("x-fapi-auth-date", OffsetDateTime.now().toString());
+        headers.put("x-fapi-auth-date", DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now(ZoneOffset.UTC)));
```
Comment (vadkor) [inline client/src/main/java/au/org/consumerdatastandards/client/api/ProtectedAPI.java:16]: Thanks for the fix!

File: client/src/main/java/au/org/consumerdatastandards/client/api/energy/EnergyAccountsAPI.java [hunk 17]
```diff
@@ -345,4 +367,6 @@ public ApiResult<EnergyBillingListResponse> listBillingForAccounts(RequestAccoun
 
         return new ApiResult<>(call.request().url().toString(), apiClient.execute(call, returnType));
     }
+
+
```
Comment (vadkor) [inline client/src/main/java/au/org/consumerdatastandards/client/api/energy/EnergyAccountsAPI.java:371]: ?

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/api/energy/EnergyApiController.java [hunk 18]
```diff
@@ -254,14 +213,23 @@ public ResponseEntity<EnergyUsageListResponse> getUsageForServicePoint(String se
...
-        data.setAccounts(Collections.singletonList(account));
+
+        Integer actualPage = getPagingValue(page, 1);
+        Integer actualPageSize = getPagingValue(pageSize, 25);
+
+        PageRequest pageRequest = PageRequest.of(actualPage - 1, actualPageSize, Sort.by(Sort.Direction.DESC, "creationDate"));
+        Page<EnergyAccountBase> energyAccountPage = service.findAccounts(openStatus, pageRequest, supportedVersion);
+        validatePageRange(actualPage, energyAccountPage.getTotalPages(), xFapiInteractionId);
+
+        data.setAccounts(energyAccountPage.getContent());
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/api/energy/EnergyApiController.java:232]: One line two many?

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountDetailResponse.java [hunk 19]
```diff
@@ -15,15 +15,15 @@
         date = "2022-01-11T14:03:27.755+11:00[Australia/Sydney]")
 public class EnergyAccountDetailResponse {
     @JsonProperty("data")
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountDetailResponse.java:17]: We also remove unneeded @JsonProperty

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/service/energy/EnergyService.java [hunk 20]
```diff
@@ -1,13 +1,8 @@
...
-import au.org.consumerdatastandards.holder.model.energy.EnergyPlan;
-import au.org.consumerdatastandards.holder.model.energy.EnergyPlanEntity;
-import au.org.consumerdatastandards.holder.model.energy.EnergyPlanDetailEntity;
-import au.org.consumerdatastandards.holder.model.energy.ParamEffective;
-import au.org.consumerdatastandards.holder.model.energy.ParamFuelTypeEnum;
-import au.org.consumerdatastandards.holder.model.energy.ParamTypeEnum;
-import au.org.consumerdatastandards.holder.repository.energy.EnergyPlanDetailRepository;
-import au.org.consumerdatastandards.holder.repository.energy.EnergyPlanRepository;
+import au.org.consumerdatastandards.holder.model.energy.*;
+import au.org.consumerdatastandards.holder.repository.energy.*;
```
Comment (vadkor) [inline data-holder/src/main/java/au/org/consumerdatastandards/holder/service/energy/EnergyService.java:4]: Wildcard import

**Review by vadkor — CHANGES_REQUESTED** [review]

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/api/energy/EnergyApiController.java (hunk 18 above)
Comment (vadkor) [reply data-holder/src/main/java/au/org/consumerdatastandards/holder/api/energy/EnergyApiController.java:232]: Nope. All good.

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/util/CdsDataLoader.java (hunk 5 above)
Comment (abeheron1) [reply data-holder/src/main/java/au/org/consumerdatastandards/holder/util/CdsDataLoader.java:111]: yes.

File: data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountDetailResponse.java [hunk 21]
```diff
@@ -15,15 +15,15 @@
         date = "2022-01-11T14:03:27.755+11:00[Australia/Sydney]")
```
[the remark this answers is not in the document]
Comment (abeheron1) [reply data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountDetailResponse.java:15]: I didn't add those, were from before

[the remark this answers is not in the document]
Comment (abeheron1) [reply data-holder/src/main/java/au/org/consumerdatastandards/holder/model/energy/EnergyAccountDetailResponse.java:15]: Oh I see what you mean, once edited manually to remove.

**Pull request merged**